cmd/kubelet/app/server.go

```diff
@@ -243,12 +243,15 @@ func (s *KubeletServer) createAPIServerClient() (*client.Client, error) {
 	if err != nil {
 		return nil, err
 	}
+	if c == nil {
+		return nil, fmt.Errorf("no client returned")
```

Comment:
No, this is one of the go landmines, where a nil interface doesn't equal an interface with a nil pointer inside it. You can leave a comment if you want.